misc/git_pick.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+
+Help="Given a Git rev-id and list of strings $0 searches \
+branch starting from rev-id for commits whose 'git log' lines contain at \
+least one string given. \
+Selected commit hashes are printed to stdout, other messages to stderr.\n\
+\n\
+Example:\n\n\
+git cherry-pick \$($0 branch_name \\\\\n\
+a_SHA1 \\\\\n\
+'author name' \\\\\n\
+'commit: message prefix' \\\\\n\
+2>>/dev/null)\n\
+"
+
+Usage="\
+usage: $0 [-h] [--help] [-[-]arg-to-git-log [..]] rev-id string [string [..]]"
+
+
+ArgsToGitLog=( )
+
+IFSBack=$IFS
+IFS=$'\n'
+while [[ "$1" =~ ^-.+$ ]]
+do
+	case "$1" in
+	"-h" )
+		echo "$Usage" 1>&2
+		echo -e "\n$Help" 1>&2
+		exit 0
+		;;
+	"--help" )
+		echo "$Usage" 1>&2
+		echo -e "\n$Help" 1>&2
+		exit 0
+		;;
+	* )
+		echo "argument to git log: $1" 1>&2
+		ArgsToGitLog+=( "$1" )
+		;;
+	esac
+	shift
+done
+IFS=$IFSBack
+
+
+if [[ 0 == $# ]]
+then
+	echo "no rev-id" 1>&2
+	echo "$Usage" 1>&2
+	exit 1
+fi
+
+LogRevId=$1
+shift
+
+
+IFSBack=$IFS
+IFS=$'\n'
+LogLines=( $(git log "${ArgsToGitLog[@]}" "$LogRevId") )
```

Comment:
With new features the script is getting closer to a regular `git log` call:
`git log -F --grep=string1 --grep=string2 --author=author1 --author=author2 --pretty=format:"%h" rev-id`
The only differences are in the order of commits (direct or reverse). The `git log` looks for the intersection of sets of `grep` and `author`, and the script looks for a union of sets.

Reply:
The script is designed to work in conjunction with cherry-pick. So, order does meter. With `ArgsToGitLog` feature may pre-filter commit sequence.